.cloudbees/testing/action.yml

```diff
@@ -116,6 +129,32 @@ runs:
           | .deploy.helm.releases[0].skipBuildDependencies = env(BUILD_DEPENDENCIES) != "true"
           | .deploy.statusCheckDeadlineSeconds = env(TIMEOUT_SECONDS)
           ' /tmp/skaffold/skaffold.yaml
+
+        if [ -s resolved_registries.out ]; then
+          cat resolved_registries.out | while read chart_location
+          do
+            chart_location=oci://$(echo "$chart_location" | awk -F':' '{print $1}') # strip the tag
+            echo "Verifying chart location: $chart_location by running helm show"
+            if helm show chart "$chart_location" --version "$CHART_VERSION"; then
+              echo "Setting chart location to $chart_location"
+              export CHART_LOCATION="$chart_location"
+              echo 'success' > /tmp/_status
+              break
+            else
+              echo 'failed' > /tmp/_status
+              continue
+            fi
+          done
+          status=$(cat /tmp/_status)
+          if [ "$status" = 'success' ]; then
+            rm resolved_registries.out /tmp/_status
+          else
+            echo "Failed to resolve any of the provided registry mirrors"
+            rm resolved_registries.out /tmp/_status
+            exit 1
+          fi
```

Comment:
When you store the `resolved_registries.out` file within the /tmp dir you don't have to clean it up here explicitly, getting rid of a branch of this conditional and therefore making the script a bit shorter.
Also, why store the loop status within a file? Can we not store that within an env var instead or just write the script so that such a state storage is not necessary and ideally split it into more readable functions?

E.g. something like this:
```sh
        findChartInMirrors() {
          if [ -s /tmp/resolved_registries.out ]; then
            cat /tmp/resolved_registries.out | while read chart_location
            do
              chart_location=oci://$(echo "$chart_location" | awk -F':' '{print $1}') # strip the tag
              echo "Verifying chart location: $chart_location by running helm show"
              if helm show chart "$chart_location" --version "$CHART_VERSION"; then
                echo "Setting chart location to $chart_location"
                export CHART_LOCATION="$chart_location"
                return 0
              fi
            done

            echo "Failed to find chart in any of the configured registry mirrors" >&2
            exit 1
          }

          findChartInMirrors
```

Reply:
Done